test/bench/custom/main.mjs

```diff
@@ -20,20 +24,71 @@ const systemInfo = () =>
   ].join('\n');
 console.log(systemInfo());
 
-const suite = new benchmark.Suite();
-
-for (const bench of benchmarks) suite.add(bench.name, bench);
-
-suite
-  .on('cycle', function logBenchmark(event) {
-    console.log(String(event.target));
-  })
-  .on('complete', function outputPerfSend() {
-    const data = Array.from(this).map(bench => ({
-      info: { test_name: bench.name },
-      metrics: [{ name: 'ops_per_sec', value: bench.hz }]
-    }));
-    console.log(util.inspect(data, { depth: Infinity, colors: true }));
-    fs.writeFileSync('customBenchmarkResults.json', JSON.stringify(data), 'utf8');
-  })
-  .run();
+function logBenchmark(event) {
+  console.log(String(event.target));
+}
+
+function processBenchmarkResult(bench, tags, metadata) {
+  return {
+    info: { test_name: bench.name },
+    metrics: [{ name: 'ops_per_sec', value: bench.hz, metadata }],
+    tags
+  };
+}
+
+let completedSuites = 0;
+function completeSuite() {
+  const metadata = { improvement_direction: 'up' };
+  if (++completedSuites >= collectedSuites.length) {
+    let cpuBaselineResults;
+    try {
+      cpuBaselineResults = JSON.parse(fs.readFileSync(`${__dirname}/../etc/cpuBaseline.json`));
```

Comment:
Optional, should be able to use await import here to get the same convenience require provides in the other benchmark. This works so no worries if you want to stay the course.